cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/CassandraClusterInfo.java

```diff
@@ -338,11 +346,14 @@ public String getVersionFromFeature()
 
     public String getVersionFromSidecar()
     {
-        LOGGER.info("Getting Cassandra versions from all nodes");
-        List<NodeSettings> allNodeSettings = Sidecar.allNodeSettingsBlocking(conf,
-                                                                             cassandraContext.getSidecarClient(),
-                                                                             cassandraContext.clusterConfig);
-        return getLowestVersion(allNodeSettings);
+        try
+        {
+            return getLowestVersion(allNodeSettings.get());
+        }
+        catch (InterruptedException | ExecutionException exception)
```

Comment:
for `InterruptedException`, it is better to restore the interrupted state for the thread. It might not be a big issue as the actually effect here is to exit the job, which makes the comment just a nit.